client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/logging/future/BucketFuture.java

```diff
@@ -0,0 +1,61 @@
+package org.kaaproject.kaa.client.logging.future;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.*;
+
+public class BucketFuture<T> implements Future<T> {
+    private static final Logger LOG = LoggerFactory.getLogger(BucketFuture.class);
+
+    private BlockingQueue<ExecutionResult<T>> queue = new ArrayBlockingQueue<>(1);
+    private volatile State state = State.WAITING;
+
+    private enum State {
+        WAITING, DONE
+    }
+
+    @Override
+    public boolean cancel(boolean mayInterruptIfRunning) {
+        throw new RuntimeException("Not implemented");
+    }
+
+    @Override
+    public boolean isCancelled() {
+        return false;
```

Comment:
No, this method is important. You can not cancel our future, thus isCancelled should always return false. It is used in executors.